The encryption-pvc-kms-vault-token canary test is failing frequently

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
As pointed out in #11741, the encryption-pvc-kms-vault-token test is failing about half the time in master.

The failure is seen during the step [validate rgw vault transit](https://github.com/rook/rook/actions/runs/4234791073/jobs/7357581690):
```
++ kubectl -n rook-ceph exec rook-ceph-rgw-my-store-a-568c9dcdb5-cwjgj -- cat /etc/vault/rgw/ssekms/vault.token
Defaulted container "rgw" out of: rgw, chown-container-data-dir (init), vault-initcontainer-token-file-setup (init)
Error from server: 
+ VAULT_TOKEN=
Error: Process completed with exit code 1.
```

Or just before the token query, another failure:
```
No resources found in rook-ceph namespace.
++ kubectl -n rook-ceph get pods -l app=rook-ceph-rgw
++ awk 'FNR == 2 {print $1}'
+ RGW_POD=
++ kubectl -n rook-ceph describe pods ''
++ grep rgw-crypt-vault-token-file
++ cut -f2- -d=
error: resource name may not be empty
+ RGW_TOKEN_FILE=
```

A successful run of this command shows the token was successfully retrieved:
```
++ kubectl -n rook-ceph exec rook-ceph-rgw-my-store-a-56b7664fd5-jh24h -- cat /etc/vault/rgw/ssekms/vault.token
Defaulted container "rgw" out of: rgw, chown-container-data-dir (init), vault-initcontainer-token-file-setup (init)
+ VAULT_TOKEN=hvs.CAESIAmxzyMzeh-aAIUGV6j8-rxz33_FSAqZbVhZMRidzU_zGh4KHGh2cy51d0RrQzFtRkNKRXV1eTU4ZzcwZ0U3dlY
```

**Expected behavior:**
Passing tests

--- a/.github/workflows/canary-integration-test.yml
+++ b/.github/workflows/canary-integration-test.yml
@@ -903,9 +903,6 @@ jobs:
           timeout 120 bash -c 'while kubectl -n rook-ceph get cephobjectstore my-store; do echo "waiting for objectstore my-store to delete"; sleep 5; done'
           kubectl create -f tests/manifests/test-object.yaml
           tests/scripts/validate_cluster.sh rgw
-          # the test is already waiting for the rgw daemon to start, but this is just a hack until we find
-          # what else we need to wait for
-          sleep 30
           tests/scripts/deploy-validate-vault.sh validate_rgw
 
       - name: collect common logs

--- a/tests/scripts/validate_cluster.sh
+++ b/tests/scripts/validate_cluster.sh
@@ -62,8 +62,12 @@ function test_demo_osd {
 }
 
 function test_demo_rgw {
-  # shellcheck disable=SC2046
-  return $(wait_for_daemon "$EXEC_COMMAND -s | grep -sq 'rgw:'")
+    timeout 360 bash -x <<-'EOF'
+    until [[ "$(kubectl -n rook-ceph get pods -l app=rook-ceph-rgw -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}')" == "True" ]]; do
+      echo "waiting for rgw pods to be ready"
+      sleep 5
+    done
+EOF
 }
 
 function test_demo_mds {

